### PR TITLE
[13.0][ADD] sale_order_import_ubl_customer_free_ref

### PR DESCRIPTION
**Pull request by TDu**
This includes #328 and builds on top of it to extract the CustomerReference information

### CODE REVIEW

File: sale_order_import/wizard/sale_order_import.py [hunk 1]
```diff
@@ -169,6 +170,20 @@ def parse_pdf_order(self, order_file, detect_doc_type=False):
     # 'doc_type': 'rfq' or 'order',
     #    }]
 
+    @api.model
+    def search_existing_order_domain(
```
Comment (simahawk) [inline sale_order_import/wizard/sale_order_import.py]: this should be private

File: sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# © 2016-2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import base64
+
+from odoo import fields
+from odoo.tests.common import TransactionCase
```
Comment (simahawk) [inline sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py]: use Savepoint + tracking_disable... you know :)

File: sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                "partner": self.env.ref("sale_order_import_ubl.fred_churchill"),
+                "shipping_partner": self.env.ref("sale_order_import_ubl.iyt"),
+                "currency": self.env.ref("base.GBP"),
+            },
+        }
+        for filename, res in tests.items():
+            f = file_open(
+                "sale_order_import_ubl_customer_free_ref/tests/files/" + filename, "rb"
+            )
+            xml_file = f.read()
```
Comment (simahawk) [inline sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py]: maybe a ctx manager?

File: sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+from odoo import api, models
+from odoo.osv.expression import AND
+
+
+class SaleOrderImport(models.TransientModel):
+    _inherit = "sale.order.import"
+
+    @api.model
+    def search_existing_order_domain(
```
Comment (simahawk) [inline sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py]: private

Comment (TDu) [reply sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py]: done

File: sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py (hunk 3 above)
Comment (TDu) [reply sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py]: done

File: sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py (hunk 2 above)
Comment (TDu) [reply sale_order_import_ubl_customer_free_ref/tests/test_ubl_order_import.py]: done

File: sale_order_import/wizard/sale_order_import.py (hunk 1 above)
Comment (TDu) [reply sale_order_import/wizard/sale_order_import.py]: done

File: sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        # Removed because will be computed
+        res.pop("client_order_ref")
+        return res
+
+    def parse_ubl_sale_order(self, xml_root):
+        res = super().parse_ubl_sale_order(xml_root)
+        ns = xml_root.nsmap
+        main_xmlns = ns.pop(None)
+        ns["main"] = main_xmlns
+        # This should be extracted !
```
Comment (simahawk) [inline sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py]: is this a TODO?
This method looks similar to the base one indeed.

Comment (TDu) [reply sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py]: A todo, a note :shrug: 
Probably something that can be done on the based module.
For next version ?

**Comment by TDu** [comment]
Rebased to remove #328 that has been merged already.

File: sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py (hunk 5 above)
Comment (simahawk) [reply sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py]: ỳes please but add a TODO ;)

Comment (TDu) [reply sale_order_import_ubl_customer_free_ref/wizard/sale_order_import.py]: @simahawk  Done

**Review by grindtildeath — APPROVED** [review]
LGTM

**Review by simahawk — APPROVED** [review]

**Pull request merged**